base/intfuncs.jl

```diff
@@ -161,8 +161,7 @@ function invmod(n::T, m::T) where T<:Integer
     g != 1 && throw(DomainError((n, m), "Greatest common divisor is $g."))
     m == 0 && throw(DomainError(m, "`m` must not be 0."))
     # Note that m might be negative here.
-    # For unsigned T, x might be close to typemax; add m to force a wrap-around.
-    r = mod(x + m, m)
+    r = (T <: Unsigned && x == typemax(x)) ? mod(x + m, m) : mod(x, m)
```

Comment:
I can't review now, but it seems this needs a `Base.hastypemax` check, not all unsigned types have `typemax`

Reply:
Thanks for your time @rfourquet . I see that all the Integer types have `typemax` defined. Correct me if I am wrong.

```
julia> intsubtypes = vcat(map(subtypes, subtypes(Integer))...)
11-element Array{Any,1}:
 BigInt 
 Int128 
 Int16  
 Int32  
 Int64  
 Int8   
 UInt128
 UInt16 
 UInt32 
 UInt64 
 UInt8  

julia> map(x -> hasmethod(typemax, Tuple{Type{x}}), intsubtypes)
11-element Array{Bool,1}:
 false
  true
  true
  true
  true
  true
  true
  true
  true
  true
  true
```

Reply:
I don't know an unsigned type without `typemax`, but it can exist and a `hastypemax` is free for the known types, so just put it, it may save us from a future bug report :)

Reply:
Oh! I will do that straightaway. Thanks :)